### ISSUE TITLE
No dry-run output for `pre-release-replacements` by default

### ISSUE THREAD

**Issue by arxanas** [crate-ci/cargo-release#451]
After #450, I tried putting `pre-release-replacements` in a `[package.metadata.release]` section for an individual package. It seems to work, but it looks like it doesn't display anything in the dry-run output:

```
$ cargo release patch
[2022-04-09T02:50:41Z ERROR cargo_release::release] Cannot release from branch "HEAD", instead switch to "*, !HEAD"
[2022-04-09T02:50:43Z WARN  cargo_release::git] Push target `origin/HEAD` doesn't exist
[2022-04-09T02:50:43Z INFO  cargo_release::release] Update git-branchless-lib to version 0.3.13
[2022-04-09T02:50:43Z INFO  cargo_release::release] Update git-branchless to version 0.3.13
[2022-04-09T02:50:43Z INFO  cargo_release::release] Publishing git-branchless-lib
    Updating crates.io index
   Packaging git-branchless-lib v0.3.12 (/Users/wkhan/workspace/git-branchless/git-branchless-lib)
   Uploading git-branchless-lib v0.3.12 (/Users/wkhan/workspace/git-branchless/git-branchless-lib)
warning: aborting upload due to dry run
[2022-04-09T02:50:46Z INFO  cargo_release::release] Publishing git-branchless
    Updating crates.io index
   Packaging git-branchless v0.3.12 (/Users/wkhan/workspace/git-branchless/git-branchless)
   Uploading git-branchless v0.3.12 (/Users/wkhan/workspace/git-branchless/git-branchless)
warning: aborting upload due to dry run
[2022-04-09T02:50:46Z WARN  cargo_release::release] Ran a `dry-run`, re-run with `--execute` if all looked good.
```

It looks like the information is produced, but at the `debug` level, so it doesn't appear in the output:

https://github.com/crate-ci/cargo-release/blob/f06a43c1b7117f7bce013ff147433883ee6aba8c/src/replace.rs#L132

I also didn't find information in the documentation on how to adjust the logging output level. I tried setting `RUST_LOG=debug`, but it didn't seem to cause a change in the output.

There are several other logs at `debug!` level, so I'm wondering if perhaps they are also being omitted? It looks like many of them would contain useful information for a dry-run.

**Comment by arxanas** [comment]
Even if we choose not to display a full diff by default, maybe we should upgrade the log level for a statement like this? I imagine that the number of output lines would be small in that case?

https://github.com/crate-ci/cargo-release/blob/ef77bee75b73ba55aa53b534078c7f98c8c26117/src/replace.rs#L76

Currently, there's no indication whatsoever that replacements are/aren't being applied.

**Comment by epage** [comment]
I think the question is "what should we generally be doing for logging".  Let's break down our high level steps:
- Version bump: info, O(crates)
- Pre-release replacements: debug, O(crates x files)
- Pre-release hook: debug, O(crates)
- Publish: info, O(crates)
- Tag: debug , O(crates)
- Post-release version bump: info, O(crates x files)
- Post-release replacements: debug, O(crates)
- Push: info, O(crates)

Thinking over this, I believe the considerations for what we log by default are:
- Don't overwhelm the user
- Make important things obvious
- Version selection is per-run so need to verify it, making it important
- Delay from publish is important
- Delay from push / authentication for push is important

The rest are more for debugging static information; we expect users set it in their config and test it at a lower frequency than releasing.

If we log replacements, we should probably log all high level steps and I'm not seeing us get enough value out of it vs users missing important information from among the flood of other information.  What we do log becomes more important as we guide the user more on what needs to be released or can be skipped within a workspace (or start doing it on the users behalf).

**Comment by dialtone** [comment]
As a brand new user, the lack of logging related to the configuration options that I specified is confusing. For example now I'm in the position where the CHANGELOG wasn't updated despite the pre-release-replacements being specified and I don't know why and there's no logging anywhere to be seen. I suppose it might be because it was the initial version of the crate, because running now with the `RUST_LOG` set will output the replacement logging lines.

IMHO rather than having a fixed stance on the debug/info level it should be based off what options the user added in the configuration.

**Comment by autarch** [comment]
FWIW, I'm facing this same problem. I'm try to get `pre-release-replacements` working and I am not able to get any useful output from the dry run.

**Comment by epage** [comment]
0.21.4 raised the log level of "Substituting" to "info" and added a `cargo release replace` subcommand which logs the diffs at "info".

I feel like this is probably the best balance we can strike for now and am going to close this

> I suppose it might be because it was the initial version of the crate, because running now with the RUST_LOG set will output the replacement logging lines.

FYI #239 #482 were tracking confusion over that and will be resolved in 0.22 (not yet released).  See #538 for details
